### ISSUE TITLE
Why cmdline completion doesn't work?

### ISSUE THREAD

**Issue by Kipok** [Kipok/NeMo-Skills#152]
Typer provides a nice utility for command line completion, but somehow it doesn't work with our scripts (at least on my side). Do we need some code fix for this?

**Comment by shtoshni** [comment]
`ns --install-completion bash/zsh/SHELL` generates the autocompletion code. 

Reload/Source the shell again.

**Comment by Kipok** [comment]
Did that work for you? Maybe something is broken in my shell setup, but for me doesn't seem to work

**Comment by shtoshni** [comment]
Which shell are you using? For zsh, I ended up copying the function to .zshrc.
Here are the new lines:

```bash
#compdef ns
_ns_completion() {
  eval $(env _TYPER_COMPLETE_ARGS="${words[1,$CURRENT]}" _NS_COMPLETE=complete_zsh ns)
}
compdef _ns_completion ns

autoload -Uz compinit; compinit
```

For bash
```bash
_ns_completion() {
    local IFS=$'
'
    COMPREPLY=( $( env COMP_WORDS="${COMP_WORDS[*]}" \
                   COMP_CWORD=$COMP_CWORD \
                   _NS_COMPLETE=complete_bash $1 ) )
    return 0
}
```

Let me know if this doesn't work.

**Comment by Kipok** [comment]
thanks, manually adding those lines seems to work, although the completion is quite slow. Do you know why it might not be doing it automatically when running that --install-completion thing?

**Comment by shtoshni** [comment]
My guess is that for zsh, how the function path is added to zshrc is not making it callable. I added a debugging message to the function call and only when I moved the function to zshrc, did I start seeing the debugging message. So my guess is maybe it has to do with executability of the function file. Regarding the speed, I'm seeing threads suggesting that loading typer itself is a bit consuming, and apparently it has to do with the `rich` library [Thread](https://github.com/fastapi/typer/discussions/744).
